examples/logging_configuration.md

```diff
@@ -0,0 +1,234 @@
+# Logging Configuration
+
+vLLM leverages Python's `logging.config.dictConfig` functionality to enable
+robust and flexible configuration of the various loggers used by vLLM.
+
+vLLM offers two environment variables that can be used, individually or in
+combination, to accommodate a range of logging configurations that vary in
+flexibility from:
+
+- vLLM's default logging configuration (least flexible)
+- coarse grained custom logging configuration
+- fine-grained custom logging configuration (most flexible)
+
+
+## Logging Configuration Environment Variables
+
+### `VLLM_CONFIGURE_LOGGING`
+
+`VLLM_CONFIGURE_LOGGING` controls whether or not vLLM propagates configuration
+from the root vLLM logger ("vllm") to other loggers used by vLLM. This
+functionality is enabled by default, but can be disabled by setting
+`VLLM_CONFIGURE_LOGGING=0` when running vLLM.
+
+If `VLLM_CONFIGURE_LOGGING` is enabled and no value is given for
+`VLLM_LOGGING_CONFIG_PATH`, vLLM will use built-in default configuration to
+configure the root vLLM logger and will use that same configuration when
+configuring other loggers used by vLLM.
+
+### `VLLM_LOGGING_CONFIG_PATH`
+
+`VLLM_LOGGING_CONFIG_PATH` allows users to specify a path to a JSON file of
+alternative, custom logging configuration that will be used instead of vLLM's
+default logging configuration. The logging configuration should be provided in
+JSON format following the schema specified by Python's [logging configuration
+dictionary schema](https://docs.python.org/3/library/logging.config.html#dictionary-schema-details).
+
+If `VLLM_LOGGING_CONFIG_PATH` is specified, but `VLLM_CONFIGURE_LOGGING` is
```

Comment:
QQ: does it have meaning in this case? (should we just assert it in code to not allow it?)

Reply:
It doesn't work this way now, but I'd like to move in the direction of leveraging `propagate=True` for descendent loggers. With that change in place, this becomes much simpler. It would boil down to:

1. If `VLLM_CONFIGURE_LOGGING` evaluates to true and `VLLM_LOGGING_CONFIG_PATH` is `None`, then apply default logging config
2. If `VLLM_LOGGING_CONFIG_PATH` is not `None`, then completely ignore `VLLM_CONFIGURE_LOGGING` and apply the given logging config

That would leave a question, should we raise an exception if `VLLM_CONFIGURE_LOGGING` evaluates to false and `VLLM_LOGGING_CONFIG_PATH` is provided? I'm okay with `VLLM_LOGGING_CONFIG_PATH` overriding `VLLM_CONFIGURE_LOGGING`, but I'm fine to raise an exception too.

To be explicit, I will add an exception if `VLLM_CONFIGURE_LOGGING` is falsy but `VLLM_LOGGING_CONFIG_PATH` is given.